### ISSUE TITLE
Add align attributes

### ISSUE THREAD

**Issue by inspier** [sharksforarms/deku#172]
It would be helpful to have `align` attributes, this is already possible in the latest version with the `pad` attributes, but requires redundant code to calculate the amount needed to align correctly. So having attributes similar to the `pad` attributes but for aligning may be nice facilities to add.

**Comment by wcampbell0x2a** [comment]
Would this use the following method? https://docs.rs/bitvec/0.22.3/bitvec/slice/struct.BitSlice.html#method.align_to

**Comment by dylanede** [comment]
I don't think that would work? That would align in _memory_, but the alignment required would be relative to the start of the whole buffer. This distinction is important because the buffer might be just bytes, and so may not start at an aligned boundary in memory. Instead what I think is needed is an extra piece of context - the current offset from the start of the buffer. That way any alignment can be calculated correctly.
